controllers/storagecluster/cephfilesystem.go

```diff
@@ -41,7 +42,17 @@ func (r *StorageClusterReconciler) newCephFilesystemInstances(initData *ocsv1.St
 		},
 	}
 
-	if initData.Spec.StorageProfiles == nil {
+	// Load StorageProfile list
+	// Get StorageProfile list in the StorageCluster's ns
+	spList := &ocsv1.StorageProfileList{}
+	spErr := r.Client.List(r.ctx, spList, &client.ListOptions{Namespace: initData.GetNamespace()})
+	if spErr != nil {
+		r.Log.Error(spErr, "Unable to list StorageProfile objects")
+	}
+	// Save StorageProfile list to be reconciled
+	r.StorageProfiles = *spList
+
+	if len(r.StorageProfiles.Items) == 0 {
```

Comment:
I think this is the wrong condition. A Lack of storage profiles does not mean that we are at a standalone deployment. Actually all of our deployments will start with no storage profile and we should not run the code in this then section for provider mode deployments

Reply:
Not accurate. Upon initialization there exists the `defaultStorageProfile`. Meaning the existence of a `StorageCluster` resource necessarily implies that there is at least that one `StorageProfile`. Might be wrong.

Reply:
Changed